device.go

```diff
@@ -158,11 +166,55 @@ func newBlockDevice(devInfo DeviceInfo) *BlockDevice {
 	}
 }
 
-func (device *BlockDevice) attach(h hypervisor) error {
+func (device *BlockDevice) attach(h hypervisor, c *Container, create bool) error {
+	device.DeviceInfo.ID = uuid.Generate().String()
+	devID := fmt.Sprintf("drive-%s", device.DeviceInfo.ID)
+
+	drive := Drive{
+		File:   device.DeviceInfo.HostPath,
+		Format: "raw",
+		ID:     devID,
+	}
+
+	// Increment the block index for the pod. This is used to determine the name
+	// for the block device in the case where the block device is used as container
+	// rootfs and the predicted block device name needs to be provided to the agent.
+	_, err := c.pod.getAndSetPodBlockIndex()
+	if err != nil {
+		return err
+	}
+
+	virtLog.Infof("Attaching block device %s", device.DeviceInfo.HostPath)
+	if create {
+		if err := h.addDevice(drive, blockDev); err != nil {
+			return err
```

Comment:
Doesn't this error path (and the one below) need to undo the effect of `c.pod.getAndSetPodBlockIndex()`?

Reply:
@jodh-intel If this errors out, then we return the error and the create or start phase simply fails. Unless we want to ignore this error and continue further, we dont need to recover and undo the effect of `c.pod.getAndSetPodBlockIndex()`

Reply:
@amshinde What happens if start fails but since the pod is created one tries to start it again ? Will the pod block index be properly set ?

Reply:
@sameo If start fails, docker calls "kill" followed by "delete" for the container process immediately if I remember correctly. And no, the pod index will not be properly set in that case.

Reply:
@amshinde you think about Docker case, but I don't think you can expect the same behavior from a CRIO case.

Reply:
Fixed in a separate commit.